qiita_db/archive.py

```diff
@@ -179,3 +181,25 @@ def retrieve_feature_values(cls, archive_merging_scheme=None,
 
             return {k: v for k, v in
                     qdb.sql_connection.TRN.execute_fetchindex()}
+
+    def insert_features(self, merging_scheme, features):
```

Comment:
A few questions, I'm having a little bit of trouble understanding this block of code. What are the different options for merging_scheme, and how does this relate back to an artifact? I'm not familiar with the Archive class and how it relates to the Artifact class and concept.

Reply:
Excellent question. This was an addition during the last release, where basically all software/plugin "decides" how their artifacts should be merged - mainly thought for bioms. This is done dynamically and based on parameters given by the plugins themself. The [main code](https://github.com/biocore/qiita/tree/master/qiita_db/software.py#L626) basically retrieves which parameters (for example, for trim the length, for close reference the reference, etc) should be accounted for when trying to merge files (meta-analysis) and we use the job/parameters that created the artifact we want to merge and the parameters of the parent (input of that job), for example: trim @ 150, close ref vs. GreenGenes. Now, here the artifact hasn't been created so we can't actually ask this directly so we query the job (processing params) and the input artifact. Note that this explanation is part of the plugins documentation; thus cc-ing, @stephanieorch and @adswafford .

Reply:
Great, thanks for the very useful explanation. I assume that the merging schemes will be described in that documentation, so this is good to go 👍